lib/fs.js

```diff
@@ -101,6 +101,10 @@ function nullCheck(path, callback) {
   return true;
 }
 
+function isFd(path) {
+  return (path >>> 0) === path && path >= 0;
```

Comment:
The original implementation does not seem to coerce at all. It errors with `TypeError: path must be a string` instead.

Nevertheless, `>>>` should always produce a number type, so the strict equality test can only ever succeed for `typeof input === 'number'`, making everything else go down the file name branch, even if that results in another `TypeError` caused by the `path` argument not being a string. This behavior remains unchanged.